### PR TITLE
backport fix/feature to stable 2.2

### PR DESCRIPTION
**Pull request by ccx1024cc**
1. #1137
b0f5edbbc708639f06f19232bcd64aca725fcf92 rafs: do not fix blob id for old bootstrap
a2ad16d4d2a1250571aafe8921285039782d4769 smoke: add `--parent-bootstrap` for merge test
7e6502711fa19fa7691138fd9e7a4da14367c9b5 builder: support `--parent-bootstrap` for merge
2. #1146
3fb31b91c29512652b5caeb8de45f02c93968a6c nydusify: forcibly enabled `--oci` option when `--oci-ref` be enabled
37e382c72d5a52caa56321c0a0efe945e2e696a5 nydusify: fix unnecessary golang-lint error
b60e92ae6adb39077a78d942170d4830197bae8e nydusify: fix `--oci` option for convert subcommand
da8083c550bdec4bf00b051aae6ae49e67703534 nydusify: fix pulling all platforms of source image
3. #1150
62d213e6faac6399a5dc705595433d9bc1c19582 rafs: fix amplify can not be skipped
4. #1163
009443b91e07a44a10b0380cc2ed340cecaccd9b nydusify: fix oci media type handle
5. #1164
d885d1a25bd5659669f77bd2b32889bec303e8f3 nydusify: cleanup work directory when conversion finish
6. #1172
6973d9db3e65fb6425bc31f872fbf3113cc7be94 fix: ci: actions are not triggered for stable/v2.2
7. #1161
edc3b09885e9c3209936a58450fe789dd4f1a63f ci test: fix bug of compiling nydus-snapshotter
8. #1123 
3031f7573aa533d4023b1785f05127c8ea335ee4 deps: bump tempfile version to 3.4.0

### CODE REVIEW

**Comment by adamqqqplay** [comment]
Maybe we can edit this PR title to make it more readable?

File: storage/Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ leaky-bucket = "0.12.1"
 libc = "0.2"
 log = "0.4.8"
 nix = "0.24"
-reqwest = { version = "0.11.11", features = ["blocking", "json"], optional = true }
+reqwest = { version = "0.11.14", features = ["blocking", "json"], optional = true }
```
Comment (adamqqqplay) [inline storage/Cargo.toml:26]: @ccx1024cc I submitted a PR earlier to fix RUSTSEC-2023-0018, but unfortunately it was not merged before v2.2.0, you could cherry-pick this commit #1123 instead.

**Review by adam3q — CHANGES_REQUESTED** [review]

File: storage/Cargo.toml (hunk 1 above)
Comment (ccx1024cc) [reply storage/Cargo.toml:26]: > 

Thanks~~~~~Missed

**Review by imeoer — APPROVED** [review]
LGTM!

**Comment by imeoer** [comment]
cc @adamqqqplay Please resolve your comment, so that we can merge it, thanks!

**Review by adam3q — APPROVED** [review]
LGTM, thanks!

**Pull request merged**